### DISCUSSION TITLE
MacOS无法使用GeoIP作为Rule

### DISCUSSION THREAD

**Discussion by monnerHenster** [eycorsican/leaf#60] [Q&A]
使用DOMAIN能够有效匹配，但是使用GEO.mmdb和GEO.dat都没法正确匹配到任何IP地址。
详细情况：
- 直接在target/debug/目录下放入geo.mmdb，然后使用GEOIP, cn , Direct,运行没有报错，但是没有匹配到任何地址。
- 直接在target/dubug/目录下放入GeoLite2-Country，然后使用EXTERNAL, mmdb:GeoLite2-Country:cn, Direct,运行报错open /Users//jessie/Rust/leaf/./target/debug/GeoLite2-Countr.mmdb failed。
- 直接在target/dubug/目录下放入GeoLite2-Country，然后使用EXTERNAL, mmdb:/Users//jessie/Rust/leaf/./target/debug/GeoLite2-Countr:cn, Direct，运行报错，open mmdb failed
![屏幕快照 2021-02-19 下午10 57 54](https://user-images.githubusercontent.com/22607651/108520877-1be62e00-7306-11eb-81aa-f2fb1bad71a4.png)

**Comment by eycorsican** [comment]
第一个你怎么确定没匹配到？你怎么确定过来的是IP而不是域名？其他两个都漏了文件后缀

**Reply by monnerHenster** [reply]
后面两个加了文件后缀，没有报错了。
我用域名和ip来访问百度首页都是没有匹配到，直接用的FINAL的代理了。百度查的ip是这个http://14.215.177.39/，中国 广东省 广州市 电信。
我目前还不清楚哪里有问题，自己先下个geoip2的解析看看mmdb是不是有损坏。

**Comment by eycorsican** [comment]
https://github.com/eycorsican/leaf/discussions/53

看是不是浏览器把IP当域名发过来了
